src/modules/DefaultDesktop.rb

```diff
@@ -25,153 +26,47 @@ def main
       @desktop = nil
 
       @initialized = false
-
-      @packages_proposal_ID_patterns = "DefaultDesktopPatterns"
     end
 
-    def MissingKey(desktop_def, key)
-      Builtins.y2warning(
-        "Wrong desktop def, missing '%1' key: %2",
-        key,
-        desktop_def.value
-      )
+    # Defaults when something in control file is not defined
+    DEFAULT_VALUES = {
+      "order"   => 99,
+      "desktop" => "unknown"
+    }
 
-      case key
-      when "order"
-        99
-      when "desktop"
-        "unknown"
-      else
-        ""
-      end
-    end
+    DEFAULT_VALUES.default_proc = Proc.new { "" }
+    DEFAULT_VALUES.freeze
 
     # Initialize default desktop from control file if specified there
     def Init
-      if @initialized == true
-        Builtins.y2debug("Already initialized")
-        return
-      end
+      return if @initialized
 
       @initialized = true
 
       # See BNC #424678
       if @all_desktops.nil?
-        Builtins.y2milestone("Getting supported desktops from control file")
+        log.info "Getting supported desktops from control file"
 
-        desktops_from_cf = []
         # supported_desktops migh be undefined
-        any_desktops_from_cf = ProductFeatures.GetFeature(
+        desktops_from_cf = ProductFeatures.GetFeature(
           "software",
           "supported_desktops"
         )
 
-        if any_desktops_from_cf != "" && any_desktops_from_cf != ""
-          desktops_from_cf = Convert.convert(
-            any_desktops_from_cf,
-            from: "any",
-            to:   "list <map>"
-          )
-        end
+        log.info "desktops from control file #{desktops_from_cf.inspect}"
+        desktops_from_cf = [] unless desktops_from_cf.is_a?(::Array)
```

Comment:
NP: I'd at least log a warning/error that the control file contains an invalid value which is ignored.

Reply:
well, main use case is nil when it is not defined. Other cases can be already seen in previous log.info and I consider it quite minor.